src/Microsoft.OData.Core/UriParser/Binders/InnerPathTokenBinder.cs

```diff
@@ -131,6 +131,7 @@ internal QueryNode BindInnerPathSegment(InnerPathToken segmentToken)
             Debug.Assert(parent != null, "parent should never be null");
 
             SingleValueNode singleValueParent = parent as SingleValueNode;
+            CollectionNavigationNode collectionParent = parent as CollectionNavigationNode;
 

```

Comment:
can we move this cast into the singleValueParent==null, to avoid the case in the case where parent is SingleValueNode?